core/chains/solana/config.go

```diff
@@ -136,6 +174,50 @@ func (c *SolanaConfig) IsEnabled() bool {
 	return c.Enabled == nil || *c.Enabled
 }
 
+func (c *SolanaConfig) SetFrom(f *SolanaConfig) {
+	if f.ChainID != nil {
+		c.ChainID = f.ChainID
+	}
+	if f.Enabled != nil {
+		c.Enabled = f.Enabled
+	}
+	setFromChain(&c.Chain, &f.Chain)
+	c.Nodes.SetFrom(&f.Nodes)
+}
+
+func setFromChain(c, f *solcfg.Chain) {
```

Comment:
These could/should be methods on the Chain, but I wanted to avoid complicating the 1.10 backport - can circle back later.